lua/neotest-dotnet/utils/build-spec-utils.lua

```diff
@@ -107,9 +108,23 @@ function BuildSpecUtils.create_specs(tree, specs, dotnet_additional_args)
     table.insert(specs, spec)
   elseif position.type == "file" then
     local proj_root = lib.files.match_root_pattern("*.csproj")(position.path)
+    local filter = {}
+    for _, child in tree:iter_nodes() do
+      local data = child:data()
+      if data.is_class then
+        table.insert(filter, "Name~" .. data.name)
```

Comment:
Unfortunately this doesn't work for XUnit. The name used in the filter isn't specific enough to work, so the dotnet test command doesn't work. What you can do is only put this filter in for NUnit framework for now. It's not ideal, but until I can come up with another more cross-framework solution for this whole adapter, that will have to do. Or if https://github.com/Nsidorenco/neotest-vstest gets to a stable point, that might end up being the adapter of choice for .net stuff.

Reply:
Updated to use the FullyQualifiedName filter for xunit which seems to work. Aside from a dodgy unit test I dont have time to fix yet.